docs/design/README.md

```diff
@@ -387,6 +416,69 @@ Directory descriptions:
 * packages: Where the RPM packages are defined.
 * variants: Where Bottlerocket image builds are defined.
 
+#### Project Variations
+
+Twoliter projects can serve one or both of two purposes.
+1) A Twoliter project may be used to create variant images.
+2) A Twoliter project may be used to create kits.
+
+Each of the above purposes may be achieved with or without depending on external kits[^nokits].
+
+[^nokits]: The use case of creating a variant without depending on external kits may seem remote.
+    But at least initially, this is what the Bottlerocket monorepo will be doing.
+
+When a user has no intention of creating a kit, it should not be necessary to define a kit in the kits directory.
+When a user has no intention of creating a variant image, it should not be necessary to define a variant in the variants directory.
```

Comment:
Do we want to support this "implicit kit" mode at the outset?

Reply:
To extend the thought - it feels like syntactic sugar or boilerplate reduction, where we could just force some local kit to be defined and use it.

The need for it may be more acute from a lack of tooling support for operations that a developer wants to do, like:
```
twoliter add <package> <kit>

# Adds a dependency from the variant to the kit that contains the package.
# This becomes more complex if we also support implicit kits.
twoliter add <package> <variant>

# Or a lower-level version that requires the developer to know where the package comes from:
twoliter add <kit> <variant>
```